### PR TITLE
📅🎨 Add the contextual menu actions base ui

### LINKED ISSUE #372: Configure the contextual menu bottom sheet
## Description
Configure the contextual menu bottom sheet view controller based on the selected item.

Actions for calendar events:
- `ContextualMenu.copyAsTimeEntryButtonTapped`
- `ContextualMenu.startFromEventButtonTapped`

Actions for new time entries:
- `ContextualMenu.copyAsTimeEntryButtonTapped`
- `ContextualMenu.discardButtonTapped`
- `ContextualMenu.editButtonTapped`
- `ContextualMenu.saveButtonTapped`

Actions for stopped time entries:
- `ContextualMenu.deleteButtonTapped`
- `ContextualMenu.editButtonTapped`
- `ContextualMenu.saveButtonTapped`
- `ContextualMenu.continueButtonTapped`

Actions for running time entries:
- `ContextualMenu.discardButtonTapped`
- `ContextualMenu.editButtonTapped`
- `ContextualMenu.saveButtonTapped`
- `ContextualMenu.stopButtonTapped`

## Dependencies
Depends on #371 

## Definition of done
- [ ] The contextual menu bottom sheet is configured for calendar events
- [ ] The contextual menu bottom sheet is configured for new time entries
- [ ] The contextual menu bottom sheet is configured for the running time entry
- [ ] The contextual menu bottom sheet is configured for stopped time entries

## Designs
Android designs TBD
You can check iOS designs meanwhile
[Light mode](https://toggl.invisionapp.com/d/main#/console/19800511/415483648/preview)
[Dark mode](https://toggl.invisionapp.com/d/main#/console/19800511/418465585/preview)

### PR DESCRIPTION
**Pull request by daividssilverio**
### Summary
This PR adds the contextual menu actions to the base contextual menu.
The layout isn't final. It will be handled at #498 .
Some actions are broken. I'll open issues to fix them.

### Technical considerations
This snippet was used because setting the tintList directly didn't work on Android API < 23:
```
        val overlayColor = item.descriptor.getColorRes(itemView.context)
        val colorFilter = PorterDuffColorFilter(overlayColor, PorterDuff.Mode.SRC_IN)
        iconCircleOverlay.background.colorFilter = colorFilter
```

### Why is it done like this and not some other way?
Feel free to make your suggestions.

### Relationships
Closes #372

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] One of our [live or file custom templates](https://github.com/toggl/android/tree/master/.idea/templates) was used in this PR

### Pointers
Review by commit. You may test it already. You should be able to see all the actions alright.
To see calendar events, you'll need to add directly to the calendar selector.

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
🦑 it

### CODE REVIEW

File: common-android/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -33,4 +33,13 @@
...
   <color name="thumb_border">#CECECF</color>
+  <color name="calendar_contextual_menu_action_start">#E57CD8</color>
+  <color name="calendar_contextual_menu_action_stop">#FF897A</color>
+  <color name="calendar_contextual_menu_action_continue">#E57CD8</color>
+  <color name="calendar_contextual_menu_action_save">#FFDE91</color>
+  <color name="calendar_contextual_menu_action_edit">#D3BAFA</color>
+  <color name="calendar_contextual_menu_action_delete">#FF897A</color>
+  <color name="calendar_contextual_menu_action_discard">#D5D0D8</color>
+  <color name="calendar_contextual_menu_action_copy">#FDE5DC</color>
+  <color name="black40percent">#65000000</color>
```
Comment (semanticer) [inline common-android/src/main/res/values/colors.xml:44]: I think this naming is to specific. We should ultimately use some names without any specific semantic meaning and re-use our colors in different contexts as much as possible. But we will probably have to sort this out globaly in some separate PR so I guess I'm fine with this for now.

Comment (daividssilverio) [reply common-android/src/main/res/values/colors.xml:44]: Agreed.

File: calendar/src/main/java/com/toggl/calendar/contextualmenu/ui/ContextualMenuActionViewHolder.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        val labels = mapOf(
+            ContextualMenuDescriptor.Start to R.string.start,
+            ContextualMenuDescriptor.Stop to R.string.stop,
+            ContextualMenuDescriptor.Continue to R.string.continue_this,
+            ContextualMenuDescriptor.Save to R.string.save,
+            ContextualMenuDescriptor.Edit to R.string.edit,
+            ContextualMenuDescriptor.Delete to R.string.delete,
+            ContextualMenuDescriptor.Discard to R.string.discard,
+            ContextualMenuDescriptor.Copy to R.string.copy_event_as_time_entry
+        )
```
[the remark this answers is not in the document]
Comment (daividssilverio) [reply calendar/src/main/java/com/toggl/calendar/contextualmenu/ui/ContextualMenuActionViewHolder.kt:65]: I'll give it a go.

File: calendar/src/main/java/com/toggl/calendar/contextualmenu/domain/ContextualMenuDescriptor.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+enum class ContextualMenuDescriptor {
+    Start,
+    Stop,
+    Continue,
+    Save,
+    Edit,
+    Delete,
+    Discard,
+    Copy,
+}
```
Comment (semanticer) [inline calendar/src/main/java/com/toggl/calendar/contextualmenu/domain/ContextualMenuDescriptor.kt:12]: This is no longer used

**Pull request merged**